components/tfx/Evaluator/with_URI_IO/component.py

```diff
@@ -0,0 +1,103 @@
+# flake8: noqa
+
+from typing import NamedTuple
+
+def Evaluator(
+    examples_uri: 'ExamplesUri',
+    model_uri: 'ModelUri',
+    output_evaluation_uri: 'ModelEvaluationUri',
+    #output_blessing_uri: 'ModelBlessingUri',
+    baseline_model_uri: 'ModelUri' = None,
+    schema_uri: 'SchemaUri' = None,
+    eval_config: {'JsonObject': {'data_type': 'proto:tensorflow_model_analysis.EvalConfig'}} = None,
+    feature_slicing_spec: {'JsonObject': {'data_type': 'proto:tfx.components.evaluator.FeatureSlicingSpec'}} = None,
+    fairness_indicator_thresholds: list = None,
+    beam_pipeline_args: list = None,
+) -> NamedTuple('Outputs', [
+    ('evaluation_uri', 'ModelEvaluationUri'),
+    #('blessing_uri', 'ModelBlessingUri'),
+]):
+    from tfx.components import Evaluator as component_class
+
+    #Generated code
+    import json
+    import os
+    import tempfile
+    import tensorflow
+    import uuid
+    from google.protobuf import json_format, message
+    from tfx.types import channel_utils, artifact_utils
+    from tfx.components.base import base_executor
+
+    arguments = locals().copy()
+
+    component_class_args = {}
+
+    for name, execution_parameter in component_class.SPEC_CLASS.PARAMETERS.items():
+        argument_value_obj = argument_value = arguments.get(name, None)
+        if argument_value is None:
+            continue
+        parameter_type = execution_parameter.type
+        if isinstance(parameter_type, type) and issubclass(parameter_type, message.Message): # Maybe FIX: execution_parameter.type can also be a tuple
+            argument_value_obj = parameter_type()
+            json_format.Parse(argument_value, argument_value_obj)
+        component_class_args[name] = argument_value_obj
+
+    for name, channel_parameter in component_class.SPEC_CLASS.INPUTS.items():
+        artifact_path = arguments.get(name + '_uri', None) or arguments.get(name + '_path', None)
+        if artifact_path:
+            artifact = channel_parameter.type()
+            artifact.uri = artifact_path + '/' # ?
```

Comment:
I think instead of a question mark, we can say: TFX components assume artifact uri is pointed to a directory, whose uri ends up with a backslash.

Reply:
Added a proper comment here.